### PR TITLE
Custom logging level && minor fixes.

### PR DESCRIPTION
**Pull request by nvthongswansea**

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -15,12 +21,30 @@ const (
 )
 
 type Client struct {
-	cfg *Config
+	cfg    *Config
+	logger *logrus.Logger
 }
```
Comment (itakouna) [inline client.go]: We do not need instance/reference of logger into Client struct. We need only a flag e.g., debuglevel, which default value is` False`. The user of the client-go lib can enable to show more debugging messages.

File: client.go [hunk 2]
```diff
@@ -15,12 +21,30 @@ const (
...
 
 func NewClient(c *Config) *Client {
+	var logLevel logrus.Level
+	if c.LogLevel == logLevel {
```
Comment (itakouna) [inline client.go]: if c.debug {
		log.SetLevel(log.DebugLevel)
	} else {
		log.SetLevel(log. ErrorLevel)
	}

Comment (itakouna) [reply client.go]: I would suggest moving this to `NewConfiguration` and read debug level form env. variable.

File: request.go [hunk 3]
```diff
@@ -70,8 +71,7 @@ func (r *Request) execute(c Client, output interface{}) error {
 	request.Header.Add("X-Auth-UserId", c.cfg.UserUUID)
 	request.Header.Add("X-Auth-Token", c.cfg.APIToken)
 	request.Header.Add("Content-Type", "application/json")
-
-	log.Printf("[DEBUG] Request body: %v", request.Body)
+	log.Infof("Request body: %v", request.Body)
```
Comment (itakouna) [inline request.go]: This should be shown only on debug level.

File: request.go [hunk 4]
```diff
@@ -84,40 +84,43 @@ func (r *Request) execute(c Client, output interface{}) error {
...
 	if result.StatusCode >= 300 {
 		errorMessage := new(RequestError) //error messages have a different structure, so they are read with a different struct
 		errorMessage.StatusCode = result.StatusCode
 		json.Unmarshal(iostream, &errorMessage)
+		log.Errorf("Error message: %v. Status: %v. Code: %v.", errorMessage.ErrorMessage, errorMessage.StatusMessage, errorMessage.StatusCode)
 		return errorMessage
 	} else {
 		json.Unmarshal(iostream, output) //Edit the given struct
-		log.Printf("[DEBUG] Response body: %v", string(iostream))
+		log.Infof("Response body: %v", string(iostream))
```
Comment (itakouna) [inline request.go]: This should be shown only on debug level.

File: client.go (hunk 2 above)
Comment (itakouna) [reply client.go]:
```
// Will log anything that is info or above (warn, error, fatal, panic). Default.
log.SetLevel(log.InfoLevel)
```

File: request.go (hunk 3 above)
Comment (nvthongswansea) [reply request.go]: Actually, it is shown only when you set the logLevel = logrus.DebugLevel (or InfoLevel). Anyway I will change the code to meet the above requirements.

**Pull request merged**